### PR TITLE
mesa: 21.3.3 -> 21.3.4

### LINKED ISSUE #153377: Mesa 21.3.x resulting in flickering on Intel GPUs
### Describe the bug

Mesa 21.3.2 present in `nixos-unstable` is causing [flickering in nheko](https://github.com/Nheko-Reborn/nheko/issues/865), and they point to [upstream bug](https://gitlab.freedesktop.org/mesa/mesa/-/issues/5782), not yet fixed as of now. Reverting to nixos-unstable  to Mesa 21.2.6 updates fixes the issue.

### Steps To Reproduce
0. Make sure you're using Intel GPU
1. Switch nixos channel to `59bfda72480`
2. Install `nheko`
3. Start `nheko`

### Expected behavior

No flickering should happen on the application screen. Mesa bug report has a video demonstrating the issue.

### Additional context
* [upstream bug in Mesa](https://gitlab.freedesktop.org/mesa/mesa/-/issues/5782)

### Notify maintainers

cc: @vcunat @primeos 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
- host os: `Linux 5.10.89, NixOS, 22.05 (Quokka)`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.5.1`
- channels(root): `"home-manager, nixos-22.05pre342525.59bfda72480"`
- channels(user): `"home-manager"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by primeos**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -47,18 +47,20 @@ self = stdenv.mkDerivation {
...
   # TODO:
   #  revive ./dricore-gallium.patch when it gets ported (from Ubuntu), as it saved
   #  ~35 MB in $drivers; watch https://launchpad.net/ubuntu/+source/mesa/+changelog
   patches = [
-    # fixes pkgsMusl.mesa build
+    # To fix flickering on Intel GPUs (iris), see https://github.com/NixOS/nixpkgs/issues/153377:
     (fetchpatch {
-      url = "https://raw.githubusercontent.com/void-linux/void-packages/b9f58f303ae23754c95d5d1fe87a98b5a2d8f271/srcpkgs/mesa/patches/musl.patch";
-      sha256 = "sha256-Jyl7ILLhn8hBJG7afnEjE8H56Wz/1bxkvlqfrXK5U7I=";
+      url = "https://gitlab.freedesktop.org/mesa/mesa/-/commit/07dc3d4238e57901ccf98e0b506d9aad2c86b9d9.diff";
```
Comment (primeos) [inline pkgs/development/libraries/mesa/default.nix:59]: Hopefully https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/14505 will be merged soon so we can ensure that this patch remains available under a fixed URL.

File: pkgs/development/libraries/mesa/default.nix [hunk 2]
```diff
@@ -47,18 +47,20 @@ self = stdenv.mkDerivation {
...
+      sha256 = "sha256-3fa1qHJes3x1/iXsxfjgy9HnEGlOyFtJatSkU1a3XDI=";
     })
+    # fixes pkgsMusl.mesa build
+    ./musl.patch
```
Comment (primeos) [inline pkgs/development/libraries/mesa/default.nix:63]: Not ideal but Void Linux is still at 21.3.3 (https://github.com/void-linux/void-packages/blob/4384718b0361de66efda66444a1526e140f88725/srcpkgs/mesa/template) and there doesn't seem to be a PR yet.

**Comment by magnetophon** [comment]
Thanks @primeos!
This fixes https://github.com/NixOS/nixpkgs/issues/153377 for me!

**Comment by vcunat** [comment]
@primeos: this sounds OK to merge to me.  Still waiting for something?  (I see the "draft" status)

**Comment by jb55** [comment]
please merge the seizures are getting worse

**Comment by vcunat** [comment]
We should ship NixOS with the usual epilepsy warnings, I guess.

**Comment by primeos** [comment]
> @primeos: this sounds OK to merge to me. Still waiting for something? (I see the "draft" status)

Yes, I was hoping for https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/14505 to be merged soon (https://github.com/NixOS/nixpkgs/pull/154926#pullrequestreview-852184906) to ensure that the referenced commit (patch URL) will not be GCed (in case the MR gets updated) but it should already be good to go (AFAIK Hydra will cache that patch anyway so it should only be an "issue" (minor annoyance) for people who don't use the official binary cache).

**Comment by vcunat** [comment]
We could look at the URL again before this hits master.   It can be resolved easily without causing any rebuild.

**Pull request merged**